Update crawls list styles

SuaYoo opened the pull request:
Resolves https://github.com/webrecorder/browsertrix-cloud/issues/608:
- Improves crawls list UI for UX and visual consistency
- Enables editing crawl metadata from the crawls list
- Upgraded Tailwind CSS

### Manual testing
1. Log in and go to crawls page
2. Regression test and verify crawl information displays as expected.
3. Scroll to completed crawl and click "..." action menu icon -> "Edit Metadata". Verify edit form is shown
4. Update form values. Verify crawl updates as expected (note: https://github.com/webrecorder/browsertrix-cloud/issues/582 is still open)
5. Regression test completed crawl actions through menu and verify crawl actions (re-run, delete, etc.) work as expected
6. Open "View" filter in control bar. Verify status icons match crawl list status icons.

### Screenshots
**View filter:**
<img width="302" alt="Screen Shot 2023-02-22 at 10 33 09 PM" src="https://user-images.githubusercontent.com/4672952/220837708-b1bddcfb-b21f-4777-9447-7d4e5abb07ba.png">

**Running & finished crawl:**
<img width="1105" alt="Screen Shot 2023-02-22 at 10 33 25 PM" src="https://user-images.githubusercontent.com/4672952/220837760-e1ec549b-83f6-4d4d-acca-4ad263726164.png">

**Hover state:**
<img width="1120" alt="Screen Shot 2023-02-22 at 10 33 45 PM" src="https://user-images.githubusercontent.com/4672952/220837832-e3672c05-426e-49f8-8d91-a91c3735e485.png">

**Running crawl action menu:**
<img width="251" alt="Screen Shot 2023-02-22 at 10 33 55 PM" src="https://user-images.githubusercontent.com/4672952/220837880-7396a12b-a18a-4165-a129-76ffb0c8852a.png">

**Completed crawl action menu:**
<img width="249" alt="Screen Shot 2023-02-22 at 10 34 20 PM" src="https://user-images.githubusercontent.com/4672952/220837914-65fd5a94-0be2-4792-80db-44c4a0c07183.png">

#### Differences from mockups
- Hover state: I adjusted the width of the action menu column so that the icon doesn't move when you hover over the crawl row.
- Crawl status filter icon color on hover: Shoelace doesn't make it easy to change just the icon color on hover, may want to refactor this down the road.
- Selected crawl status filter tag style: Again, Shoelace doesn't make it easy to customize this right now.

SuaYoo committed:
add list styles

SuaYoo committed:
adjust columns

SuaYoo committed:
update headings

SuaYoo committed:
add dropdown items

SuaYoo committed:
add crawl status component

SuaYoo committed:
show status

SuaYoo committed:
wip update status dropdown

SuaYoo committed:
update status icon

SuaYoo committed:
update action menu icons

SuaYoo committed:
update color

SuaYoo committed:
update action methods

SuaYoo committed:
move metadata dialog

SuaYoo committed:
update metadata from crawls list

SuaYoo committed:
adjust dropdown style

SuaYoo committed:
update dialog state

SuaYoo committed:
link to crawl

SuaYoo committed:
fix navigation

SuaYoo committed:
show first seed name

SuaYoo committed:
update loading state

SuaYoo committed:
update hover style

SuaYoo committed:
update duration and page

SuaYoo committed:
update page stats

SuaYoo committed:
update statuses

SuaYoo committed:
align footer

stavares843 reviewed (commented):
<img width="1252" alt="Captura de ecrã 2023-02-23, às 19 22 42" src="https://user-images.githubusercontent.com/29093946/221009266-b7056110-7a2c-432a-b65c-4ae321aa2b61.png">


very minor but I wonder if the colors should be the same across the project, example the badge here in starting status is purple but in the dropdown before that view is green

SuaYoo commented:
> <img alt="Captura de ecrã 2023-02-23, às 19 22 42" width="1252" src="https://user-images.githubusercontent.com/29093946/221009266-b7056110-7a2c-432a-b65c-4ae321aa2b61.png">
> 
> very minor but I wonder if the colors should be the same across the project, example the badge here in starting status is purple but in the dropdown before that view is green

Good catch, I’ll update this

Shrinks99 commented:
+1 to unifying the status indicator colours

Icon changes for stopped and timed out are neat!

I noticed switching to `transform: scale(1.02);` caused way more problems than it's worth for the dropdown position, lets keep this implementation!  Also kinda neat how it actually displays more of the title! :)

---

The "Go to Crawl Config" button should be a shortcut to the latest version of the config it was spawned from, not the config tab of the crawl object.  This is intended to replace the "Edit Config" button that we currently have.  If that config has been deleted this should display a popup like we do with the "Re Run Crawl" option allowing users to duplicate the config so they may run it again.

---

It would be good to expose the full name with a `title=""` attribute so users can see the full name if they hover over it.  Helpful when truncated!

<img width="479" alt="Screenshot 2023-02-23 at 9 29 10 PM" src="https://user-images.githubusercontent.com/5672810/221077294-3576b98c-4bdd-4aba-8f12-ea5e59aa2b17.png">

SuaYoo committed:
update crawl detail state

SuaYoo committed:
update link to crawl config

stavares843 reviewed (commented):
<img width="523" alt="Captura de ecrã 2023-02-24, às 20 30 52" src="https://user-images.githubusercontent.com/29093946/221285466-202fe942-781e-44d9-8476-397f2a63dd44.png">

<img width="550" alt="Captura de ecrã 2023-02-24, às 20 30 38" src="https://user-images.githubusercontent.com/29093946/221285475-ef04edfb-0df8-4a11-9302-0eb06d046c0c.png">

other minor thing was the first dropdown when hovering the icons are still colored but in the options of each crawl, the icon is white, should be white in both dropdown's when hovering?

SuaYoo committed:
handle longer title

SuaYoo commented:
> The "Go to Crawl Config" button should be a shortcut to the latest version of the config it was spawned from, not the config tab of the crawl object.

Updated! (cc @stavares843)

<img width="1123" alt="Screen Shot 2023-02-24 at 12 16 46 PM" src="https://user-images.githubusercontent.com/4672952/221288909-ea218107-328b-44fe-bf15-15e76000d20b.png">

Note that the "Back" link doesn't go back to crawls, I've created https://github.com/webrecorder/browsertrix-cloud/issues/641 to address this.

> If that config has been deleted this should display a popup like we do with the "Re Run Crawl" option allowing users to duplicate the config so they may run it again.

Unsure of what popup you're referring to @Shrinks99, do you mean the "Crawl is already running" alert? Configs with a crawl cannot be deleted, only deactivated. If this menu item is intended to link to the crawl config detail page, we should show a persistent instructional message on the inactive crawl config detail page that suggests duplicating the config--rather than blocking navigation to the crawl config page in order to show an alert on the crawls list based on an extra HTTP call (or return the config state from the backend API.)

> It would be good to expose the full name with a `title=""` attribute so users can see the full name if they hover over it. Helpful when truncated!

Text truncated with `text-overflow: ellipsis;` is already accessible in that it's read by screen readers. The [title attribute actually is not accessible](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title#accessibility_concerns) for the use case you're suggesting. This brings up a good point, though, the truncate was actually a bit of a lazy copy-paste from the "Config Author" column and I've reverted it to not truncate the crawl name, in parity of what we have now.

<img width="1096" alt="Screen Shot 2023-02-24 at 12 43 29 PM" src="https://user-images.githubusercontent.com/4672952/221288802-b4ac9922-178e-44c2-aa90-390c3ffbdaf7.png">

Open to other suggestions on how to handle longer text in columns.

If we want to show the full config author name on hover I think we could come up with a tooltip that's useful for both truncated and not truncated name--maybe the full, untruncated name, date the config was created, and the crawl schedule if scheduled?

SuaYoo commented:
> other minor thing was the first dropdown when hovering the icons are still colored but in the options of each crawl, the icon is white, should be white in both dropdown's when hovering?

I made a note of this in the PR description `Shoelace doesn't make it easy to change just the icon color on hover`, I think we could address this in a follow up PR since the dropdown requires some refactoring anyway to re-style the selected statuses per mockups?

stavares843 commented:
ohh was that 😂 thanks for clarifying! 🔨

Shrinks99 commented:
On truncation: This text should be truncated, aware it's already good with screen readers (I've actually started some testing with them!), would like to handle this with a popup which is why I suggested `title` though I think you're right and this is an area better visually handled by a Shoelace tooltip rather than a native one.  In the future we may also be providing some additional data with a similar tooltip over the status indicator as well!

Names should also be truncated!  In the future I'd like to make these into tags?  See [mockup](https://www.figma.com/file/PBmDTlRUAkx5eNc3wMhLdB/Crawls-Page-V1?node-id=574%3A5722&t=4RbA2bSSwxnRyNje-1) if you're curious.

---


> Unsure of what popup you're referring to

This one!

<img width="474" alt="Screenshot 2023-02-24 at 4 34 35 PM" src="https://user-images.githubusercontent.com/5672810/221297815-6f2f7a2d-f3ce-47ce-84b4-e14f39d9588c.png">

This ~~popup~~ notification appears when a crawl config has been deactivated and you try to re-run the crawl.  It helpfully directs users to an auto populated config editor where they can essentially create what is to them a new config and run it again.  This is the same workflow I'd like to implement when navigating to disabled configs (which currently lets them view the disabled config, something we don't let them do anywhere else).  If they want to view the config settings for that crawl they should look inside the crawl object.

This will allow users to accomplish the action of editing the config which is what they should be trying to do if they hit it.  The only reason we can't call it "edit config" is that configs are a whole seperate object that lives elsewhere. 🙃

Shrinks99 reviewed (approved):
Spinning out my previous requests into issues, everything else looks great here :D
